app/scripts/views/tour/tour-page-controller.js

```diff
@@ -42,6 +47,15 @@
 
             ctl.onPrevClick = onPrevClick;
             ctl.onNextClick = onNextClick;
+
+            Visualization.get(Config.ids.tour).then(onVisReady, onVisError);
```

Comment:
The catch for `getLayers` failing if the layers aren't there yet is the expected condition (initialization not having finished yet) that I'm talking about; that sort of thing is generally a code smell.  Yes, I understand you're doing this to have it only created once.  What I'm suggesting is that if you pass in `onVisReady` and `onVisError` to `add`, you don't need to do the initialization work with `get`. `get` can be used then only to retrieve an already-initialized object you want to reference elsewhere.